再々PR：ゲームを超最低限一周できるようにする

134n opened the pull request:
### 概要
ゲームを超最低限一周できるようにする

※間違えてPRを削除してしまったため再PRの再PR😭　原因https://github.com/134n/2D_TapGame/pull/11#issuecomment-2395336554
bb5d8f812656847666431d0f373225adacdc4a2f からの更新を見てほしい
https://github.com/134n/2D_TapGame/pull/11#issue-2550547369 前回のPR(消してしまったやつ)
### 詳細

**1. タイトルの追加
2. ゲームシステムの変更
3. リザルトの追加**

### やった事
1. タイトルの追加

> ボタン.OnClickAsObservable() : ボタンを押したら、
> SceneManager.LoadScene("Main");　: メインに遷移させる

2. ゲームシステムの変更
> float型の変数を毎秒1ずつ減算して0になったら、
> SceneManager.LoadScene("Result");　: リザルトに遷移させる

3. リザルトの追加

> ボタン.OnClickAsObservable() ボタンを押したら、
> SceneManager.LoadScene("Main");　: メインに遷移させる


https://github.com/user-attachments/assets/6ebab35c-4946-4ef0-ab54-fcb987c544d1

134n committed:
Clean : Mainフォルダの作成

スクリプトをシーン別で分けたい

134n committed:
Add : タイトルの実装

ボタンを押したらメインへ遷移する

134n committed:
Add : カウントダウンの実装

10秒カウントしたらリザルトへ遷移する

134n committed:
Add : リザルトの実装

ボタンを押したらタイトルへ遷移

134n committed:
clean : クラス名の変更

134n committed:
clean : シーンをconstで定義

134n committed:
Clean : カウントダウンするコードのリファクタリング

UniRxを用いてコードをわかりやすく

134n committed:
Update : スタートボタンにStartButtonUI.csをアタッチ

134n committed:
Update : タイトルボタンにTitleButtonUI.csをアタッチ

134n committed:
Clean：変数名の変更

Title→titleButton

134n committed:
Clean：コードのリファクタリング

間隔を空けて読みやすく

134n committed:
Clean：シーン移動ボタンのクラス名をわかりやすく

134n committed:
Clean：const変数の命名をわかりやすく

134n committed:
Clean：CountDown.csのリファクタリング

mokapants commented on Assets/Scripts/Result/TitleButtonUI.cs:
```diff
@@ -3,7 +3,7 @@
 using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 
-public class TitleButtonUI : MonoBehaviour
```
TitleButtonUIファイルのクラス名がGoToTitleButtonになってて
GoToTitleButtonファイルのクラス名がTitleButtonUIになってるので
動かなさそうな気がしているんだけど動作確認はできてる？？

134n replied:
それぞれの同名クラスが別ファイルに存在していたので、コンパイルエラーにならず動作していました。
各クラスとファイルの名前が一致するように修正いたしました。

mokapants commented on Assets/Scripts/Title/StartButtonUI.cs:
```diff
@@ -3,7 +3,7 @@
 using UnityEngine.UI;
 using UnityEngine.SceneManagement;
 
-public class StartButtonUI : MonoBehaviour
```
こちらもしかり、ファイル名とクラス名がちぐはぐになっていそうー

134n replied:
修正いたしました！

mokapants commented on Assets/Scripts/Main/CountDown.cs:
```diff
@@ -14,10 +14,7 @@ private void Start()
-            SceneManager.LoadScene(LoadResultScene);
-        })
-        .AddTo(this);
+            .Subscribe(_ =>SceneManager.LoadScene(ResultSceneName))
```
空白をきれいに入れておきたいですー
`(_ => Xxxx)`

134n replied:
修正いたしました！

134n committed:
Update : GoToMainButton.csのリファクタリング

ファイル名 = スクリプト名になるように変更。StartButtonUIのクラス名がGoToMainButtonでコードが同じだったため動作確認はできていた。クラス名をStartButtonUI→GoToMainButtonに変更した際にちぐはぐになってしまった。

134n committed:
Update : GoToTitleButton.csのリファクタリング

ファイル名 = スクリプト名になるように変更。TitleButtonUIのクラス名がGoToTitleButtonでコードが同じだったため動作確認はできていた。クラス名をTitleButtonUI→GoToTitleButtonに変更した際にちぐはぐになってしまった。

134n committed:
Clean：CountDown.csのリファクタリング

読みやすいように空白を正しく入れる
